core/nodepool/config/templates/stack-template.json

```diff
@@ -326,7 +326,7 @@
                   "Action": [
                     "s3:GetObject"
                   ],
-                  "Resource": "arn:aws:s3:::{{$.UserDataWorkerS3Path}}"
+                  "Resource": "arn:aws{{if .IsChinaRegion}}-cn{{end}}:s3:::{{$.UserDataWorkerS3Path}}"
```

Comment:
nit but `func (r Region) Partition() string` to return `aws-cn` or `aws` according to http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html would be slightly better.